Source/JavaWrappers.cpp

```diff
@@ -274,6 +281,97 @@ namespace java::lang
     }
 }
 
+namespace java::websocket
+{
+    WebSocketClient::WebSocketClient(std::string url, std::function<void()> open_callback, std::function<void()> close_callback, std::function<void(std::string)> message_callback, std::function<void()> error_callback)
+        : Object{android::global::GetWebSocketClass()}
+        , m_openCallback{std::move(open_callback)}
+        , m_messageCallback{std::move(message_callback)}
+        , m_closeCallback{std::move(close_callback)}
+        , m_errorCallback{std::move(error_callback)}
+    {
+        static JNINativeMethod methods[] = {
+                {"closeCallback", "()V", (void*)OnClose },
+                {"openCallback", "()V", (void*)&WebSocketClient::OnOpen },
+                {"messageCallback", "(Ljava/lang/String;)V", (void*)&WebSocketClient::OnMessage },
+                {"errorCallback", "()V", (void*)&WebSocketClient::OnError },
+        };
+        m_env->RegisterNatives(m_class, methods, 4);
+
+        // initialize socket
```

Comment:
nit
```suggestion
        // Construct the socket
```

Or just remove the comment since it doesn't really add much.